### PR TITLE
Add chat-agent sample

### PR DESCRIPTION
**Pull request by isidorn**

### CODE REVIEW

**Comment by isidorn** [comment]
fyi @mjbvz for cat ideas :)

File: chat-agent-sample/extension.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
+"use strict";
```
Comment (roblourens) [inline chat-agent-sample/extension.js:1]: Don't commit this and the .map file

File: chat-agent-sample/package.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    "categories": [
+        "Other"
+    ],
+    "activationEvents": [
+        "onStartupFinished"
+    ],
+    "enabledApiProposals": [
+        "chatAgents2",
+        "chatRequestAccess",
+        "interactive",
```
Comment (roblourens) [inline chat-agent-sample/package.json:26]: You shouldn't need this proposal

File: chat-agent-sample/package.json [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        "readonlyMessage"
```
Comment (roblourens) [inline chat-agent-sample/package.json:27]: It doesn't look like this one is used

File: chat-agent-sample/src/extension.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+const CAT_LEARN_SYSTEM_PROMPT = 'You are a cat! Your job is to explain computer science concepts in a funny manner of a cat. Always start your response by stating what concept you are explaining.';
+const CAT_PLAY_SYSTEM_PROMPT = 'You are a cat that wants to play! Reply in a helpful way for a coder, but with the hidden meaning that all you want to do is play.';
+const MEOW_COMMAND_ID = 'cat.meow';
+
+export function activate(context: vscode.ExtensionContext) {
+
+	const teachResult = { /* you can return anything in your result object */ };
+    const playResult = { /* you can return anything in your result object */ };
+    // Define a Cat chat agent handler. 
+    const handler: vscode.ChatAgentHandler = async (request: vscode.ChatAgentRequest, context: vscode.ChatAgentContext, progress: vscode.Progress<vscode.ChatAgentProgress>, token: vscode.CancellationToken) => {
```
Comment (roblourens) [inline chat-agent-sample/src/extension.ts:12]: I think it should do something even without a slash command (I could hack on this a bit)

File: chat-agent-sample/src/extension.ts [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+            const access = await vscode.chat.requestChatAccess('copilot');
+            const messages = [
+                {
+                    role: vscode.ChatMessageRole.System,
+                    content: CAT_PLAY_SYSTEM_PROMPT
+                },
+            ];
+            const request = access.makeRequest(messages, {}, token);
+            for await (const fragment of request.response) {
+                const incomingText = fragment.replace('[RESPONSE END]', '');
```
Comment (roblourens) [inline chat-agent-sample/src/extension.ts:46]: Do you actually need this line?

File: chat-agent-sample/src/extension.ts [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+                progress.report({ content: incomingText });
+            }
+            return teachResult;
+        } else if (request.slashCommand?.name == 'play') {
+            const access = await vscode.chat.requestChatAccess('copilot');
+            const messages = [
```
Comment (roblourens) [inline chat-agent-sample/src/extension.ts:38]: Was this intended to use the user's query?

File: chat-agent-sample/extension.js (hunk 1 above)
Comment (isidorn) [reply chat-agent-sample/extension.js:1]: Thanks. Done

File: chat-agent-sample/package.json (hunk 2 above)
Comment (isidorn) [reply chat-agent-sample/package.json:26]: Done. Thanks

File: chat-agent-sample/package.json (hunk 3 above)
Comment (isidorn) [reply chat-agent-sample/package.json:27]: Thanks. Done

File: chat-agent-sample/src/extension.ts (hunk 4 above)
Comment (isidorn) [reply chat-agent-sample/src/extension.ts:12]: I can do this tomorrow - will have more time then.

File: chat-agent-sample/src/extension.ts (hunk 5 above)
Comment (isidorn) [reply chat-agent-sample/src/extension.ts:46]: Good point. Removing it.

File: chat-agent-sample/src/extension.ts (hunk 6 above)
Comment (isidorn) [reply chat-agent-sample/src/extension.ts:38]: Yeah, let's use the user's query here.

**Review by roblourens — APPROVED** [review]

**Pull request merged**